newIDE/app/src/MainFrame/index.js

```diff
@@ -1818,14 +1828,20 @@ const MainFrame = (props: Props) => {
 
       saveUiSettings(state.editorTabs);
 
-      const storageProviderOperations = getStorageProviderOperations(
+      // Remember the old storage provider, as we may need to use it to get access
+      // to resources.
+      const oldStorageProvider = getStorageProvider();
+      const oldStorageProviderOperations = getStorageProviderOperations();
+
+      // Get the methods to save the project using the *new* storage provider.
+      const newStorageProvider = getStorageProvider();
```

Comment:
I think you need to define `newStorageProvider` after calling the new storage provider operations since it's the second instruction that changes the `currentStorageProvider` in `ProjectStorageProviders`

Reply:
Good catch, thanks!